All classes integrated into the UI

georgeadadii opened the pull request:
I.Adadi - R.Schulz - F.Clancy - Added all the classes to the UI and integrated them for seamless program execution - 21:37 - 08/04/2024

georgeadadii committed:
All classes integrated into the UI

I.Adadi - R.Schulz - F.Clancy - Added all the classes to the UI and integrated them for seamless program execution - 21:37 - 08/04/2024

postolav commented on UI Development/gui.pde:
```diff
@@ -14,6 +14,12 @@ class Gui {
   String splitLine[];
   String linesSplit[][];
+    
+  Airport airport;
+  Date date;
+  Delays delay;
+  ChartPage charts = new ChartPage();
+  Scroll scroll = new Scroll();
```
For these declarations, see the comment made on the encapsulation of our program.

postolav commented on UI Development/gui.pde:
```diff
@@ -51,6 +57,9 @@ class Gui {
         linesSplit[rowIndex][coloumnIndex] = tmpSplitLine[coloumnIndex];
       }
     }
+    airport = new Airport(linesSplit);
+    date = new Date();
+    delay = new Delays();
```
For these instantiations, see comment made on the encapsulation of our program.

postolav commented on UI Development/gui.pde:
```diff
@@ -172,6 +181,23 @@ class Gui {
       }
     }
+    
+    if (searchFilters.size() != 0 )
+    {
+      if (searchFilters.contains("airports"))
+      {
+        airport.draw();
+      }
+      if (searchFilters.contains("dates"))
+      {
+        date.draw();
+      }
+      if(searchFilters.contains("delays"))
+      {
+        delay.draw();
+      }
+    }
```
This if-block should practically be put in the ChartPage class, as it deals with the back-end.

For example, notice how no button or searchbar object is drawn in the Gui top-level class. Instead, both types of objects are drawn in the `draw()` of the Page they're on. This has been done for the purpose of encapsulation. Gui must **only** take care of drawing the pages.

Applying the same principle, any back-end class used for the purpose of charting should have its `draw()` called within the `draw()` of the ChartPage class.

postolav commented on UI Development/chartPage.pde:
```diff
@@ -1,12 +1,204 @@
+    super.setHeaderFont("PPSupplyMono-Regular.otf", 20);
+  }
 
   public void draw() {
```
Where all the back-end class objects (Airports, Dates, Delays) should be drawn. Insert the if-block I commented on in this function.

postolav reviewed (changes requested):
Took a glance at the code and suggested some changes, let me know if the proposals I'm making in the comments are not possible to implement.

postolav commented:
### Partial Error Handling
I noticed that user input error handling has been built into _delays.pde_, which shouldn't be the case. All error input must be dealt with in the _SearchPage_ class. Such is for the purpose of properly dealing with the `inputFilters` ArrayList.

**Recommended course of action:**
Take out the error handling built into _delays.pde_, have it later implemented in a future commit.

georgeadadii committed:
Full-Stack Reconfiguration

Items Completed:
- Configured the backend architecture to have connection points with the frontend counterpart.
- The ResultsPage (renamed from ChartsPage) frontend class now calls the Airports, Dates & Delays backend classes.
- This alteration overhauls previous iterations of the program, where all the backend classes were being called in the top-level GUI class.
- The reconfiguration has also fixed a bug where the searchbar textfields would continue to show on the succeeding ResultsPage objects.
- The purpose of our reconfiguration was to strengthen our program encapsulation.

Future Objectives:
- Working towards the implementation of error handling for each searchbar.

20:00 | 9/04/2024 | I.Adadi

georgeadadii committed:
Updated Error Handling and Chart Methods

I.Adadi - Implemented some error handling, changed the pie chart method and changed the bar chart method also. - 20:44 - 09/04/2024

postolav committed:
Merged Latest Error Handling With Fullstack Reconfig

Items Completed:
- The latest error handling changes made by George in his ui-with-search-queries branch has been merged with the fullstack-reconfig branch.

Future Objectives:
- Fixing a bug where the back buttons don't show on-screen for the ResultsPage objects.
- Fixing a bug where the bar chart x-axis labels are overlapping one another.
- Fixing a bug with the pie chart.

23:00 | 9/04/2024

georgeadadii committed:
Error Handling and Refactoring

I.Adadi - Handled all the errors/wrong input possibilities for all the classes and added comments and headers to code - 16:34 - 10/04/2024

georgeadadii committed:
Refactoring for Query Classes

I.Adadi - Refactoring for query classes completed - 18:19 - 10/04/2024

postolav committed:
Compiled Fullstack Implementations

Items Completed:
- Completed the refactoring for every one of our classes.
- Ensured for fluid program execution one last time.

Future Objectives:
- Finalizing the program for submission.

19:00 | 10/04/2024

georgeadadii committed:
Additional Configs
